vividus-plugin-excel/src/main/java/org/vividus/bdd/steps/ExcelFileSteps.java

```diff
@@ -0,0 +1,90 @@
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.jbehave.core.annotations.When;
+import org.jbehave.core.model.ExamplesTable;
+import org.vividus.bdd.context.IBddVariableContext;
+import org.vividus.bdd.variable.VariableScope;
+import org.vividus.util.ResourceUtils;
+
+public class ExcelFileSteps
+{
+    private final IBddVariableContext bddVariableContext;
+
+    public ExcelFileSteps(IBddVariableContext bddVariableContext)
+    {
+        this.bddVariableContext = bddVariableContext;
+    }
+
+    /**
+     * Creates temporary xlsx file with specified content and puts path to that file to variable with specified name.
+     * Created file will be removed while termination of the JVM
+     * @param content Any valid ExamplesTable that will be written to the file
+     * @param scopes The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of variable's scope<br>
+     * <i>Available scopes:</i>
+     * <ul>
+     * <li><b>STEP</b> - the variable will be available only within the step,
+     * <li><b>SCENARIO</b> - the variable will be available only within the scenario,
+     * <li><b>STORY</b> - the variable will be available within the whole story,
+     * <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     * </ul>
+     * @param variableName name of path variable
+     * @throws IOException if an I/O exception of some sort has occurred
+     */
+    @When("I create temporary excel file with content:$content and put path to $scopes variable "
+            + "with name `$variableName`")
+    public void initVariableUsingExcelFilePath(ExamplesTable content,
+                 Set<VariableScope> scopes, String variableName) throws IOException
+    {
+        Path pathTemporaryFile = ResourceUtils.createTempFile("name", ".xlsx", null);
+        try (XSSFWorkbook workbook = new XSSFWorkbook();
+                FileOutputStream fileOut = new FileOutputStream(pathTemporaryFile.toFile()))
+        {
+            XSSFSheet sheet = workbook.createSheet();
+            fillData(content, sheet);
+            workbook.write(fileOut);
+        }
+        bddVariableContext.putVariable(scopes, variableName, pathTemporaryFile);
+    }
+
+    private void fillData(ExamplesTable content, XSSFSheet sheet)
+    {
+        Row rowHeaders = sheet.createRow(0);
+        List<String> headers = content.getHeaders();
+        int columnSize = headers.size();
+        IntStream.range(0, columnSize).forEach(index ->
+            rowHeaders.createCell(index).setCellValue(headers.get(index)));
+        IntStream.rangeClosed(1, content.getRows().size()).forEach(rowIndex -> {
+            Row sheetRow = sheet.createRow(rowIndex);
+            IntStream.range(0, columnSize).
+                      forEach(colIndex ->
+                          sheetRow.createCell(colIndex).setCellValue(content.getRowAsParameters(rowIndex - 1, true).
```

Comment:
row.createCell(index).setCellValue(value) logic can be moved into separate func and reused

Reply:
And I propose to move this logic into a separate class with an interface like createExcel(List<Map<String, String> data, Path toAFile)